<?xml version="1.0" encoding="utf-8"?><testsuites><testsuite errors="1" failures="0" hostname="jcwang" name="pytest" skipped="0" tests="1" time="0.435" timestamp="2024-07-29T15:09:43.845223"><testcase classname="cslam" file="" name="" time="0.000"><error message="collection failure">/usr/lib/python3/dist-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:83: in &lt;lambda&gt;
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
/usr/lib/python3/dist-packages/_pytest/python.py:187: in pytest_collect_file
    return ihook.pytest_pycollect_makemodule(path=path, parent=parent)
/usr/lib/python3/dist-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:83: in &lt;lambda&gt;
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
/opt/ros/foxy/lib/python3.8/site-packages/launch_testing/pytest/hooks.py:188: in pytest_pycollect_makemodule
    entrypoint = find_launch_test_entrypoint(path)
/opt/ros/foxy/lib/python3.8/site-packages/launch_testing/pytest/hooks.py:181: in find_launch_test_entrypoint
    module = path.pyimport()
/usr/lib/python3/dist-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
/usr/lib/python3/dist-packages/_pytest/assertion/rewrite.py:304: in load_module
    exec(co, mod.__dict__)
tests/test_algebraic_connectivity.py:6: in &lt;module&gt;
    from cslam.algebraic_connectivity_maximization import AlgebraicConnectivityMaximization, EdgeInterRobot
cslam/algebraic_connectivity_maximization.py:5: in &lt;module&gt;
    from cslam.mac.mac import MAC
cslam/mac/mac.py:13: in &lt;module&gt;
    from cslam.mac.utils import *
cslam/mac/utils.py:9: in &lt;module&gt;
    import numba
/usr/local/lib/python3.8/dist-packages/numba/__init__.py:55: in &lt;module&gt;
    _ensure_critical_deps()
/usr/local/lib/python3.8/dist-packages/numba/__init__.py:40: in _ensure_critical_deps
    raise ImportError(msg)
E   ImportError: Numba needs NumPy 1.22 or greater. Got NumPy 1.17.</error></testcase></testsuite></testsuites>